core/publishers.go

```diff
@@ -12,18 +12,31 @@ func Publishers(publishMsgs <-chan models.Message, config *models.Config) {
 	log.Print("Initializing Publishers")
 	for {
 		message := <-publishMsgs
-		for _, route := range message.Routes {
-			if match, _ := parse.Match(route.Match, message.Out.Text+" "+message.Out.Detail); match || route.Match == "*" {
-				for _, connector := range config.Connectors {
-					if connector.Active {
-						if sendToConnector(connector.ID, route.Connectors) {
-							if connector.Debug {
-								log.Print("Broadcasting to " + connector.ID + " (type:" + connector.Type + ") for route " + route.Connectors)
-								log.Printf("Message: %+v", message)
-								log.Print("")
+		for _, route := range config.Routes {
+			for _, m := range route.Matches {
+				match := true
+				match, _ = parse.Match(m.Message, message.Out.Text+" "+message.Out.Detail)
```

Comment:
I did fix it over the weekend to pass in the match by reference so as to read slightly better.